src/pykiso/lib/connectors/cc_pcan_can.py

```diff
@@ -399,115 +418,87 @@ def _merge_trc(self) -> None:
                 # If a log file is not provided, take the fist trace created as result file
                 result_trace = list_of_traces[0]
             else:
-                # Else write the first trace content in the log file and then remove it
+                # Rename the trace if necessary
                 result_trace = Path(self.trace_path / self.trace_name)
-                list_of_traces[0] = shutil.move(
-                    str(list_of_traces[0]), str(result_trace)
-                )
-                list_of_traces[0] = Path(list_of_traces[0])
-
-            # End of the trace header
-            first_message_line = 33
 
-            # Get start time of the first trc file
-            with list_of_traces[0].open("r") as trc:
-                data = trc.read().splitlines(True)
-
-                first_trc_start_time = CCPCanCan._get_trace_start_time(
-                    data, first_message_line
+                list_of_traces[0] = Path(
+                    shutil.move(str(list_of_traces[0]), str(result_trace))
                 )
-                message_idx = len(data[first_message_line:])
 
-            list_of_traces.pop(0)
+            # Extract header
+            header_data = CCPCanCan._extract_header(list_of_traces[0])
 
-            # Append all trace files
-            for file in list_of_traces:
-                with file.open("r") as trc:
-                    data = trc.read().splitlines(True)
+            # Only keep first trace header
+            with result_trace.open("r") as trc:
+                data = trc.read().splitlines(True)
 
-                    # Get offset between current and first trc file
-                    trc_start_time = CCPCanCan._get_trace_start_time(
-                        data, first_message_line
-                    )
-                    offset = trc_start_time - first_trc_start_time
+            # Get messages from all traces
+            messages = self._read_trace_messages(list_of_traces, result_trace)
 
-                    # Fix message number and time offset inconstistancies
-                    corrected_data = CCPCanCan._fix_merge_inconsistencies(
-                        data[first_message_line:], offset, message_idx
-                    )
-                    message_idx += len(data[first_message_line:])
+            # Write header in result trace
+            with result_trace.open("w") as trc:
+                for line in data:
+                    if line[0] == ";":
+                        trc.write(line)
```

Comment:
```suggestion
            result_trace.write_text(header_data)
```
Wouldn't this do exactly the same?